app/services/catalog/create_request_for_applied_inventories.rb

```diff
@@ -0,0 +1,34 @@
+module Catalog
+  class CreateRequestForAppliedInventories
+    attr_reader :order
+
+    def initialize(id)
+      @order = Order.find_by!(:id => id)
+      @item = @order.order_items.first
+    end
+
+    def process
+      send_request_to_compute_applied_inventories
+
+      @order.update(:state => "Waiting for inventories")
+      self
+    end
+
+    private
+
+    def send_request_to_compute_applied_inventories
+      service_plan = TopologicalInventoryApiClient::AppliedInventoriesParametersServicePlan.new(
+        :service_parameters => @item.service_parameters
+      )
+      TopologicalInventory.call do |api|
+        task_id = api.applied_inventories_for_service_offering(service_offering_ref, service_plan).task_id
+
+        @item.update(:topology_task_ref => task_id)
```

Comment:
Shoul `@order.update(:state => "Waiting for inventories")` be set without knowing if the `task_id` comes back? Possibly might want that under a transaction?

Reply:
Would it make more sense to simply update the order after the item has been updated with the task id?

Reply:
You're correct. This is fine, we would never get back to setting the `:message` on the order if the order item wasn't updated.